sdk/lib/opentelemetry/sdk/trace/trace_state.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      # TraceState represents the OpenTelemetry::Trace::Tracestate values that
+      # are part of the OpenTelemetry ecosystem, contained in a single entry
+      # using the `ot` key. The format and handling requirements are specified
+      # at https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-handling.md
+      class TraceState
```

Comment:
The overloading of `TraceState` here made me scratch my head for awhile. We're really defining a programmatic way to interact with the part of the API TraceState that corresponds to the special, reserved `ot` bits.

It's a little wordy, but it would be more clear to me if we called it `SDK::Trace::TraceState::OpenTelemetryConcern` or something like that, so it's clear that it's a subset. And then you don't end up with interesting calls like `ot = OpenTelemetry::SDK::Trace::Tracestate.from_tracestate(tracestate)` 😄

(Part of me almost wants to suggest that we could somehow fold all of this in as an SDK extension to the API Tracestate class - intercepting calls to get / set the `ot` key, etc ... but that feels equally problematic somehow).

Reply:
Yeah - the overloading is unpleasant. Again, I'm trying to adhere closely to the spec terms. `OpenTelemetryConcern` is reasonable. The spec uses that term in describing the `ot` [Key](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-handling.md#key).